src/nwb_benchmarks/core/_base_network_benchmark.py

```diff
@@ -0,0 +1,102 @@
+"""Base class for implementing benchmarks for evaluating network performance metrics for streaming read of NWB files."""
+
+import os
+import time
+
+from asv_runner.benchmarks.mark import SkipNotImplemented
+
+from ._capture_connections import CaptureConnections
+from ._network_profiler import NetworkProfiler
+from ._network_statistics import NetworkStatistics
+
+
+class BaseNetworkBenchmark:
+    """
+    Base class for ASV Benchmark network performance metrics for streaming data access.
+
+    Child classes must implement:
+        1) The `operation_to_track_network_activity_of`, which replaces the usual time_ or mem_ operation being tracked.
+        2) The `setup_cache` method to explicitly recieve the necessary keyword arguments to be passed to the operation.
+    """
+
+    def operation_to_track_network_activity_of(self, **keyword_arguments):
+        raise SkipNotImplemented()
+
+    def setup_cache(self, **keyword_arguments) -> dict:
+        """
+        Strategy essentially mimics a single benchmark execution (setup + run).
+
+        Only supports single operations (ignores `repeat` and prohibits `params`).
+        """
+        self.start_net_capture()
+        t0 = time.time()
+        self.setup(**keyword_arguments)
+        self.operation_to_track_network_activity_of(**keyword_arguments)
+        t1 = time.time()
+        total_time = t1 - t0
+        self.stop_netcapture()
+        self.net_stats["network_total_time"] = total_time
+        return self.net_stats
+
+    def start_net_capture(self):
+        # start the capture_connections() function to update the current connections of this machine
+        self.connections_thread = CaptureConnections()
+        self.connections_thread.start()
+        time.sleep(0.2)  # not sure if this is needed but just to be safe
+
+        # start capturing the raw packets by running the tshark commandline tool in a subprocess
+        self.netprofiler = NetworkProfiler()
+        self.netprofiler.start_capture()
+
+    def stop_netcapture(self):
```

Comment:
```suggestion
    def stop_net_capture(self):
```

We should use the same naming convention here as for `start_net_capture`, i.e., either with or without the `_` between `net` and `capture` . Probably my oversight, but something we should fix before we merge.

Reply:
Fixed, as well as some other improved readability, in #25